schemas/pom.xml

```diff
@@ -51,8 +51,29 @@
     <module>dublin-core</module>
     <module>iso19110</module>
     <module>iso19139</module>
-    <module>iso19115-3.2018</module>
   </modules>
+
+  <profiles>
+    <profile>
+      <id>iso19115-3.2018</id>
+      <activation>
+        <file><exists>iso19115-3.2018</exists></file>
+      </activation>
+      <modules>
+        <module>iso19115-3.2018</module>
+      </modules>
+    </profile>
+    <profile>
+      <id>hnap</id>
+      <activation>
```

Comment:
Are you adding the hnap schema plugin profile to the geonetwork core?  Any reason this would be added but not the other schema plugins from metadata 101? Was this used for testing and accidentally added to this pull request?

Reply:
I am using this as an example. One benefit of this approach is that all the modules can Ben in profiles. No need to build locally once they have been updated to include plugin folder and deploy to OSGeo repo. 

I felt including an example was useful at this time, and it helped me write this docs.

Reply:
My personal preference is to remove it as I believe there are already too many profiles and I personally found it difficult to identify the meaning of all these profiles when I was starting with the code.  I believe "iso19115-3.2018" should be the example as everyone will probably know what that means.  I don't believe many will understand what hnap is?

If we add hnap then maybe we should add them all (or all the active ones) - but it could end up creating a large pom file?  If they are added then it will definitely make it easier for us to build hnap as we will just need to specify the profile.

If we are going to add these then there should be a better naming standard so that it is clear that this is a plugin. i.e. maybe plugin-iso19115-3.2018 and plugin-iso19139.ca.HNAP... So that if we add others then it is clear that these profiles are for adding plugins

Reply:
Good discussion @ianwallen, thank you for your feedback. I will use `iso19115-3.2018` as the example of selectively enabling a profile if the submodule is added. I think the instructions could move to `web/pom.xml` since they are now all about that folder.

I agree with your feedback, and would like to implement it.

1. Adding profiles to `web/pom.xml` for each schema is way to keep the build the same for everyone.

2. It is possible to enable profiles in ~/.m2/settings.xml if you wish to always build with the same set of schema plugins:

```
<settings>
  ...
  <activeProfiles>
      <activeProfile>iso19139.ca.HNAP</activeProfile>
    </activeProfiles>
```

The community benefit is we all end up working on the same geonetwork project, rather than being forced to fork :) (This community building aspect is important to me and one reason I am working on the build system and shared osgeo repository)

Reply:
Regarding adding the profile to web/pom.xml, isn't that the purpose of the add_schema.sh? It is a way to add the standard expected code to the pom file for cases where you may want it.  I agree that if it is already there, then it is easy to turn it off and on but I don't think it is reasonable to expect that we will have the full list of schemas in the pom files. This is why I was suggesting that we (I can do this) create  a add_schema_profile.sh that would allow the user to add the standard profile settings.  You did mention an option using ant in an email - I did not fully understand this option so I'm waiting to see a sample of this as this may be a better option.

.m2 modification is no an option for us as this would not be easily portable to build env as well as getting developers to add this settings locally will cause issues.  Also I have several build folder on my pc and regularly create others for testing - in some cases, I build hnap and other cases I don't and I don't want to hit debug issues related to this being activated and I had forgotten.

Reply:
I understand about `.m2` modification to list activeProfiles not working for you.

Can I ask how many profiles you usually work with? There are [two recommended](https://stackoverflow.com/questions/4766396/how-to-specify-active-profiles-in-maven3) ways of managing this:
- Use active by default (this does not work across teams)
- Use of settings.xml

If you *must* fork I would prefer if you fork and add:
```
    <activation>
      <activeByDefault>true</activeByDefault>
    </activation>
```
To your required profiles.

However I would like to check if the work I have done already takes away this need?

- Checkout the metadata101 plugins you need to into schemas folder
- Make sure they are added to a schemas/.gitignore to avoid accidentally commiting
- A profile in schemas/pom.xml will check for the folder being present and include the module
- A profile in web/pom.xml will check for the folder being present and include the dependency and unpack the artifact

This should address the issue for you to one of initial setup, and avoid having to make any pom.xml changes to work with you desired schema plugins.

Reply:
The only reason this would work for us is because hnap module is pre-added.  I don't believe we are trying to set it up just so I don't have to fork?  For this to work with the full geonetwork community, we would need to have all possible profiles in the pom.xml which I don't believe is feasible.   If I wanted to avoid a fork, then I would simply add the hnap post deployment.

Geonetwork seems to require lots of config changes that can only be done with a fork. We currently need to do a fork just to change the spring security in our deployment.   It may take time before all these config options can be configured based on one war file without doing post deployment changes. Hopefully we can make it there someday...

Lots of documentation states to say away from `<activeByDefault>` - in our case we generally setup the deployment so that it works by default without having to modify profiles.  So in this case we would be adding the hnap schema to the pom file either manually as a small configuration change or we will use the one that it pre-installed in the pom if it is already there.

Reply:
I expect to add all metadata101 profiles here over time: goal is not to require forking.